tests/test_provenance_dataset.py

```diff
@@ -0,0 +1,303 @@
+"""Tests for DatasetModel."""
+
+from datetime import date
+from pydantic import HttpUrl, ValidationError
+import pytest
+from geojson_pydantic import Point
+
+from bento_lib.discovery.models.ontology import OntologyTerm
+from bento_lib.discovery.models.provenance import (
+    DatasetModel,
+    Contact,
+    Count,
+    License,
+    Logo,
+    Organization,
+    ParticipantCriteria,
+    Publication,
+    SpatialCoverageFeature,
+    SpatialCoverageProperties,
+)
+
+
+def test_dataset_model(basic_pi, basic_contact):
+    """Test complete DatasetModel."""
+    institution = Organization(
+        name="Test University",
+        description="Research institution",
+        contact=Contact(email=["contact@test.edu"], address=None, phone=None),
+        roles=["Institution"],
+        grant_number=None,
+    )
+
+    funder = Organization(
+        name="National Science Foundation",
+        description=None,
+        contact=basic_contact,
+        roles=["Funder"],
+        grant_number="NSF-12345",
+    )
+
+    dataset = DatasetModel(
+        schema_version="1.0",
+        title="Test Study",
+        description="A comprehensive test study",
+        dataset_id="test-study-001",
+        keywords=["genomics", "cancer"],
+        stakeholders=[basic_pi, institution, funder],
+        spatial_coverage="Canada",
+        version="1.0",
+        privacy="Controlled Access",
+        license=License(
+            label="CC BY 4.0", type="Creative Commons", url=HttpUrl("https://creativecommons.org/licenses/by/4.0/")
+        ),
+        counts=[Count(count_entity="participants", value=100, description="Total participants")],
+        primary_contact=basic_pi,
+        links=[],
+        publications=[
+            Publication(
+                title="Study Results",
+                url=HttpUrl("https://doi.org/10.1234/test"),
+                doi="10.1234/test",
+                publication_type="Journal Article",
+                authors=None,
+                publication_date=None,
+                publication_venue=None,
+                description=None,
+            )
+        ],
+        data_access_links=[HttpUrl("https://example.com/data")],
+        release_date=date(2023, 1, 1),
+        last_modified=date(2023, 6, 1),
+        participant_criteria=[ParticipantCriteria(type="Inclusion", description="Adults 18+")],
+        pcgl_domain=["Cancer"],
+        pcgl_status="ONGOING",
+        pcgl_context="RESEARCH",
+        pcgl_program_name="Cancer Genomics Program",
+    )
+
+    assert dataset.title == "Test Study"
+    assert len(dataset.stakeholders) == 3
+    assert dataset.pcgl_status == "ONGOING"
+    assert dataset.pcgl_context == "RESEARCH"
+    assert dataset.pcgl_domain[0] == "Cancer"
+
+
+def test_dataset_model_with_custom_domain(basic_pi):
```

Comment:
a lot of repeated code with small changes, could use dictionary expansion + model validate to reduce duplicate code.